clients/ops/admin-ui/src/features/datastore-connections/add-connection/manual/DSRCustomizationForm.tsx

```diff
@@ -0,0 +1,163 @@
+import { Box, HStack, Text, VStack } from "@fidesui/react";
+import { useAlert } from "common/hooks";
+import { TrashCanSolidIcon } from "common/Icon";
+import { FieldArray, Form, Formik } from "formik";
+import { useRouter } from "next/router";
+import React from "react";
+import * as Yup from "yup";
+
+import { STEPS } from "../constants";
+import CustomInput from "../forms/CustomInput";
+import { ButtonGroup as ManualButtonGroup } from "./ButtonGroup";
+import { Field } from "./types";
+
+type DSRCustomizationFormProps = {
+  data: Field[];
+  isSubmitting: boolean;
+  onSaveClick: (values: any, actions: any) => void;
+};
+
+const DSRCustomizationForm: React.FC<DSRCustomizationFormProps> = ({
+  data = [],
+  isSubmitting = false,
+  onSaveClick,
+}) => {
+  const router = useRouter();
+  const { errorAlert } = useAlert();
+
+  const handleCancel = () => {
+    router.push(STEPS[1].href);
```

Comment:
We can get into an interesting state here where the form has been partially filled out and then cancelled but there are still resources created.

If the user filled in the Connector parameters and a ConnectionConfig was created, but then they make it to the DSR customization screen and hit cancel, a ConnectionConfig will be created, but no ManualWebhook will exist(this stores the fields).  

No fields set up for this one:
![Screen Shot 2022-09-14 at 12 22 40 PM](https://user-images.githubusercontent.com/9755598/190221592-9b0731ab-875a-4036-a95c-83759d625a52.png)

I will make some backend changes around this, perhaps the "test" for a manual webhook should verify that fields exist, and exclude webhooks without fields from holding up privacy request execution. 

But from a frontend perspective, is there more we could do to prevent this state, or at least alert to this state?

Reply:
Let's discuss this partial state scenario given different user workflows.

Reply:
Note from a backend perspective, I've added "test" functionality which fails if the manual webhook is only partially setup here. Additional edit functionality could allow the user to update the webhook and add the fields.